### PR TITLE
Fill segments using bedtools

### PR DESCRIPTION
**Pull request by Kdreval**
This is a different implementation of fill_segments script that is reliant on bedtools. It serves the same goal of filling missing segments in seg files to make them compatible with downstream tools, but this task is achieved using bedtools subtract feature and some perl file modifications.

### CODE REVIEW

File: fill_segments/1.0/fill_segments.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# Run bedtools substract to find regions that are missing from seg file.
+# The perl pipe part will assign the neutral segment log.ratio and no LOH for the missing segments.
+# For the missing segments, +1 is added to start position and 1 is substracted from the end position
+# to ensure they do not overlap with original segments of seg file by 1 position.
+# 0.00 are explicitly used to easily identify segments supplemented by this script
+bedtools subtract -a $ARM_BED_PATH -b $RESULTS_PATH.headerless.bed | perl -lane '@a=split;$a[1] = ++$a[1];$a[2] = --$a[2]; $a[3]="0.00"; $a[4]="0.00"; $a[5]=$ENV{THIS_SAMPLE_ID}; print join "\t", @a;' > $RESULTS_PATH.temp
+
+# Merge the initial seg file with the missing segments and rearrange columns to match the style of seg files
+cat $RESULTS_PATH.headerless.bed $RESULTS_PATH.temp | sort -V | perl -pale 'BEGIN { $"="\t"; } $_ = "@F[$#F,0..$#F-1]"' > $RESULTS_PATH.merged.seg
```
[the remark this answers is not in the document]
Comment (Kdreval) [reply fill_segments/1.0/fill_segments.sh]: Thanks Jasper! This all has been addressed and pushed to the branch 👍

**Review by lkhilton — CHANGES_REQUESTED** [review]
I think my only request is that the chromArm.hg19.bed file should actually be called chromArm.grch37.bed for consistency wrt genome_build or projection wildcards. Otherwise looks great! Glad @Jwong684 was able to share his experiences to handle edge cases with you.

**Review by Jwong684 — APPROVED** [review]
Seems like it should work (not sure about the perl parts though)

**Comment by Kdreval** [comment]
> I think my only request is that the chromArm.hg19.bed file should actually be called chromArm.grch37.bed for consistency wrt genome_build or projection wildcards. Otherwise looks great! Glad @Jwong684 was able to share his experiences to handle edge cases with you.

Thank you, yes the naming was not correct - I updated those files to reflect the proper build 👍

**Review by Jwong684 — DISMISSED** [review]

**Review by lkhilton — APPROVED** [review]

**Pull request merged**